artemis-server/src/main/java/org/apache/activemq/artemis/core/server/transformer/ServerMessageImpl.java

```diff
@@ -0,0 +1,190 @@
+    * This is to avoid setMessageTransient which set the address without changing the buffer
+    *
+    * @param address
+    */
+   @Override
+   public void forceAddress(SimpleString address) {
+      message.setAddress(address);
+   }
+
+   @Override
+   public int incrementRefCount() throws Exception {
+      throw new UnsupportedOperationException();
+   }
+
+   @Override
+   public int decrementRefCount() throws Exception {
+      throw new UnsupportedOperationException();
+   }
+
+   @Override
+   public int incrementDurableRefCount() {
+      throw new UnsupportedOperationException();
+   }
+
+   @Override
+   public int decrementDurableRefCount() {
+      throw new UnsupportedOperationException();
+   }
+
+   @Override
+   public ServerMessage copy(long newID) {
+      throw new UnsupportedOperationException();
+   }
+
+   @Override
+   public ServerMessage copy() {
+      throw new UnsupportedOperationException();
+   }
+
+   @Override
+   public int getMemoryEstimate() {
+      return message.getMemoryEstimate();
+   }
+
+   @Override
+   public int getRefCount() {
+      return message.getRefCount();
+   }
+
+   @Override
+   public ServerMessage makeCopyForExpiryOrDLA(long newID,
+                                               MessageReference originalReference,
+                                               boolean expiry,
+                                               boolean copyOriginalHeaders) throws Exception {
+      throw new UnsupportedOperationException();
+   }
+
+   @Override
+   public void setOriginalHeaders(ServerMessage otherServerMessage, MessageReference originalReference, boolean expiry) {
+
+      ICoreMessage other = otherServerMessage.getICoreMessage();
+
+      SimpleString originalQueue = other.getSimpleStringProperty(Message.HDR_ORIGINAL_QUEUE);
+
+      if (originalQueue != null) {
+         message.putStringProperty(Message.HDR_ORIGINAL_QUEUE, originalQueue);
+      } else if (originalReference != null) {
+         message.putStringProperty(Message.HDR_ORIGINAL_QUEUE, originalReference.getQueue().getName());
+      }
+
+      if (other.containsProperty(Message.HDR_ORIG_MESSAGE_ID)) {
+         message.putStringProperty(Message.HDR_ORIGINAL_ADDRESS, other.getSimpleStringProperty(Message.HDR_ORIGINAL_ADDRESS));
+
+         message.putLongProperty(Message.HDR_ORIG_MESSAGE_ID, other.getLongProperty(Message.HDR_ORIG_MESSAGE_ID));
+      } else {
+         message.putStringProperty(Message.HDR_ORIGINAL_ADDRESS, new SimpleString(other.getAddress()));
+
+         message.putLongProperty(Message.HDR_ORIG_MESSAGE_ID, other.getMessageID());
+      }
+
+      // reset expiry
+      message.setExpiration(0);
+
+      if (expiry) {
+         long actualExpiryTime = System.currentTimeMillis();
+
+         message.putLongProperty(Message.HDR_ACTUAL_EXPIRY_TIME, actualExpiryTime);
+      }
+
+      // TODO ASk clebert
+      //message.bufferValid = false;
```

Comment:
@mtaylor remove it... message.putStringProperty, setExpiration... or any other change will set bufferValid = false on the delegation.. so you can safely remove this.. and remove this TODO